src/rgw/rgw_rest_s3.cc

```diff
@@ -3523,18 +3523,20 @@ RGWOp *RGWHandler_REST_Service_S3::op_post()
 RGWOp *RGWHandler_REST_Bucket_S3::get_obj_op(bool get_data)
 {
   // Non-website mode
-  int list_type = 1;
-  s->info.args.get_int("list-type", &list_type, 1);
-
-   // Non-website mode    // Non-website mode
   if (get_data) {   
-    if (list_type == 1) {
-       return new RGWListBucket_ObjStore_S3;     
-    } else if(list_type == 2) {
-      return new RGWListBucket_ObjStore_S3v2;
-    } } else {
-    return new RGWStatBucket_ObjStore_S3;    
-  }   }
+    int list_type = 1;
+    s->info.args.get_int("list-type", &list_type, 1);
+    switch (list_type) {
+      case 1:
+        return new RGWListBucket_ObjStore_S3;
+      default:
```

Comment:
this is missing two important lines from the initial pr:
```c++
      case 2:
        return new RGWListBucket_ObjStore_S3v2;
```

Reply:
Thanks @cbodley ! When I started backporting this, https://github.com/ceph/ceph/pull/29849 hasn't been merged yet, so I assumed I had to remove any reference to v2. I'll amend this.

Reply:
Thanks @cbodley ! When I started backporting this, https://github.com/ceph/ceph/pull/29849 hasn't been merged yet, so I assumed I had to remove any reference to v2. I'll amend this.